### PR TITLE
calculate distance outside of lap timing

### PR DESCRIPTION
**Pull request by brentpicasso**
allows distance to be calculated before lap timing starts. Feedback is that users expect this to happen.

### CODE REVIEW

File: src/lap_stats/lap_stats.c [hunk 1]
```diff
@@ -446,6 +444,8 @@ static void lapstats_location_updated(const GpsSnapshot *gps_snapshot)
     const GeoPoint *gp = &gps_snapshot->sample.point;
     const float target_radius = degrees_to_meters(config->TrackConfigs.radius);
 
+    update_distance(gps_snapshot);
+
```
Comment (stieg) [inline src/lap_stats/lap_stats.c]: So they get distance if they haven't configured a start finish.  Seems silly to me.  Thoughts?

Comment (brentpicasso) [inline src/lap_stats/lap_stats.c]: feedback is it should start calculating distance even if we're not in a lap - but resets once we start the beginning of a lap.

**Comment by stieg** [comment]
Change seems sane.  Add a unit test please to ensure that this works as expected.

**Comment by brentpicasso** [comment]
added unit test to check for distance increasing before current lap increments.

**Pull request merged**